src/modules/iasql_functions/rpcs/iasql_get_sql_since.ts

```diff
@@ -0,0 +1,342 @@
+ */
+async function jsonValueToDbString(
+  tableName: string,
+  key: string,
+  value: any,
+  modsIndexedByTable: { [key: string]: ModuleInterface },
+  orm: TypeormWrapper,
+): Promise<string> {
+  if (value === undefined || typeof value === 'string' || typeof value === 'object') {
+    return getDbString(value);
+  }
+  // If `value`'s type is `number` we might need to recreate a sub-query because it could be an `id` referencing other table.
+  // In this case we need to get Typeorm metadata for this `tableName` and inspect columns and relations and recreate the sub-query if necessary.
+  // We need to recreate the sub-query because database `id` columns will not be the same across databases connected to the same cloud account.
+  const metadata = await getMetadata(tableName, modsIndexedByTable, orm);
+  if (value && typeof value === 'number' && metadata) {
+    // If `metadata instanceof EntityMetadata` means that there's an Entity in Typeorm which it's table name is `tableName`
+    if (metadata instanceof EntityMetadata) {
+      const keyRelationMetadata = metadata.ownColumns
+        .filter(oc => oc.databaseName === key && oc.referencedColumn?.databaseName === 'id')
+        .map(oc => oc.relationMetadata)
+        ?.pop();
+      if (keyRelationMetadata) {
+        return await recreateSubQuery(
+          value,
+          keyRelationMetadata.inverseEntityMetadata,
+          modsIndexedByTable,
+          orm,
+        );
+      }
+    }
+    // If `metadata instanceof RelationMetadata` means that there's no Entity in Typeorm which it's table name is `tableName`,
+    // but theres a join table linking entities and `tableName` is that join table. In this case we need to check `joinColumns`
+    // which will have the columns from the owner of the relationship and `inverseJoinColumns` will have the columns coming from
+    // the other entities in the relationship.
+    if (metadata instanceof RelationMetadata) {
+      const keyRelationMetadata = metadata.joinColumns
+        .filter(jc => jc.databaseName === key && jc.referencedColumn?.databaseName === 'id')
+        .map(jc => jc.relationMetadata)
+        ?.pop();
+      if (keyRelationMetadata) {
+        return await recreateSubQuery(value, keyRelationMetadata.entityMetadata, modsIndexedByTable, orm);
+      }
+      const keyInverseRelationMetadata = metadata.inverseJoinColumns
+        .filter(jc => jc.databaseName === key && jc.referencedColumn?.databaseName === 'id')
+        .map(jc => jc.relationMetadata)
+        ?.pop();
+      if (keyInverseRelationMetadata) {
+        return await recreateSubQuery(
+          value,
+          keyInverseRelationMetadata.inverseEntityMetadata,
+          modsIndexedByTable,
+          orm,
+        );
+      }
+    }
+  }
+  return `${value}`;
+}
+
+/** @internal */
+function getDbString(v: any): string {
+  if (v === undefined) return `${null}`;
```

Comment:
Why this syntax to get `'null'`?

Reply:
because we treat `undefined` as `null`s in the database, this should not occur, but if the value coming from the JSON is `undefined` we updated it to `null`. This function just returns the valid "typed" strings for the query.